mlir/include/mlir/Dialect/Ptr/IR/PtrDialect.td

```diff
@@ -63,6 +64,54 @@ def Ptr_PtrType : Ptr_Type<"Ptr", "ptr", [
       return $_get(memorySpace.getContext(), memorySpace);
     }]>
   ];
+  let extraClassDeclaration = [{
+    // `PtrLikeTypeInterface` interface methods.
+    /// Returns `Type()` as this pointer type is opaque.
+    Type getElementType() const {
+      return Type();
+    }
+    /// Clones the pointer with specified memory space or returns failure
+    /// if an `elementType` was specified or if the memory space doesn't
```

Comment:
This is confusing, what's the point of providing `elementType` if it always triggers a failure?

Reply:
Nevermind, I thought I was reading the declaration for the method on the Interface!